### ISSUE TITLE
Open file dialog -- can't paste path in clipboard

### ISSUE THREAD

**Issue by lee321987** [geany/geany#4010]
In most (all?) other programs when I open the "Open file..." dialog I can instantly paste the file path I have in the clipboard and hit enter, but with geany when i try to paste I get:
> "The folder contents could not be displayed.
> Operation not permitted"
I can press CTRL+L to show a field where pasting works, but this extra CTRL+L step isn't needed with other programs.
Anyone know why or how I can change this behavior?

**Comment by elextr** [comment]
AFAIK there is no way of making it work that with the entry visible by default unless somebody can find it in the [docs](https://docs.gtk.org/gtk3/class.FileChooserDialog.html)

**Comment by techee** [comment]
Maybe could https://github.com/geany/geany/pull/3861 help here?

@lee321987 You'll need to build Geany from the latest git master to try but then there's Edit->Preferencess->Miscellaneous->Use platform-native file dialogs you could check. That said, I'm not sure if it helps - on my Debian Xfce it displays the same file open dialog but if you say other applications support this, it's possible that Ubuntu uses some different file-opening dialog by default.

**Comment by elextr** [comment]
Nah, any Linux GTK desktop is still the same [expletive deleted] GTK Linux open dialog and has the same problem.

Not sure if the KDE open is used or it uses GTK dialog on those desktops?

Otherwise the native filechooser only changes on Win and Mac.

**Comment by lee321987** [comment]
I'm guessing it's realated to this GTK bug:
https://bugs.launchpad.net/ubuntu/+source/meta-gnome3/+bug/1830979

Long story short:  don't hold your breath for a fix.

I just tested it on a bunch of programs, and I can paste immediately into all these program's file open dialogs:

> Mousepad, SciTE, Nomacs, Firefox (non-Snap), VLC



I can't paste immediately with any of these:

> Geany, Ristretto, GIMP, Libre Office

I wish I knew exactly what determines whether a program will have the bug.
Thanks for the comments guys.

**Comment by eht16** [comment]
Mousepad does not have this behavior and so I got curious what could be the difference. It turned out `gtk_file_chooser_set_local_only(FALSE)` is the reason.

Though I don't get exactly why GTK behaves likes this. I had a quick look at the GTK filechooser code and it seems to happen somewhere after https://gitlab.gnome.org/GNOME/gtk/-/blob/main/gtk/gtkfilechooserwidget.c?ref_type=heads#L5304 where the paste is handled.
But in the widget code I didn't find a relation to the local_only mode, so I guess one has to dive deeper into the filechooser internals and understand this beast. I did not.

https://gitlab.gnome.org/GNOME/gtk/-/issues/5872 could be related however in my tests I could not even paste a folder path successfully, so it might also be a different issue. Anyway, the report didn't receive any attention at all :(.
